src/ray/raylet/worker_pool.cc

```diff
@@ -1043,27 +1070,30 @@ void WorkerPool::PushWorker(const std::shared_ptr<WorkerInterface> &worker) {
 }
 
 void WorkerPool::TryKillingIdleWorkers() {
-  const auto now = get_time_();
+  const absl::Time now = get_time_();
 
   // Filter out all idle workers that are already dead and/or associated with
   // jobs that have already finished.
   int64_t num_killable_idle_workers = 0;
+  auto worker_killable = [now](const IdleWorkerEntry &entry) -> bool {
+    return entry.keep_alive_until < now;
+  };
+
+  // First, kill must-kill workers: dead ones, job finished ones. Also calculate killable
+  // worker count.
   for (auto it = idle_of_all_languages_.begin(); it != idle_of_all_languages_.end();) {
-    const auto &idle_worker = it->first;
-    if (idle_worker->IsDead()) {
+    if (it->worker->IsDead()) {
       it = idle_of_all_languages_.erase(it);
       continue;
     }
 
-    const auto &job_id = idle_worker->GetAssignedJobId();
-    if (finished_jobs_.contains(job_id)) {
+    const auto &job_id = it->worker->GetAssignedJobId();
+    if (finished_jobs_.count(job_id) > 0) {
```

Comment:
nit: I feel `contains()` is more clear than `count() > 0`

Reply:
done